pkg/client/cache/deploymentconfig.go

```diff
@@ -0,0 +1,95 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/labels"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deployutil "github.com/openshift/origin/pkg/deploy/util"
+)
+
+// StoreToDeploymentConfigLister gives a store List and Exists methods. The store must contain only deploymentconfigs.
+type StoreToDeploymentConfigLister struct {
+	cache.Indexer
+}
+
+// Exists checks if the given deploymentconfig exists in the store.
+func (s *StoreToDeploymentConfigLister) Exists(dc *deployapi.DeploymentConfig) (bool, error) {
+	_, exists, err := s.Indexer.Get(dc)
+	return exists, err
+}
+
+// List all deploymentconfigs in the store.
+func (s *StoreToDeploymentConfigLister) List() ([]*deployapi.DeploymentConfig, error) {
+	configs := []*deployapi.DeploymentConfig{}
+	for _, c := range s.Indexer.List() {
+		configs = append(configs, c.(*deployapi.DeploymentConfig))
+	}
+	return configs, nil
+}
+
+// GetConfigForController returns the managing deployment config for the provided replication controller.
+func (s *StoreToDeploymentConfigLister) GetConfigForController(rc *kapi.ReplicationController) (*deployapi.DeploymentConfig, error) {
+	dcName := deployutil.DeploymentConfigNameFor(rc)
+	obj, exists, err := s.Indexer.GetByKey(rc.Namespace + "/" + dcName)
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, fmt.Errorf("deployment config %q not found", dcName)
+	}
+	return obj.(*deployapi.DeploymentConfig), nil
+}
+
+func (s *StoreToDeploymentConfigLister) DeploymentConfigs(namespace string) storeDeploymentConfigsNamespacer {
+	return storeDeploymentConfigsNamespacer{s.Indexer, namespace}
+}
+
+type storeDeploymentConfigsNamespacer struct {
```

Comment:
Lack of a `Get` is weird.  Also, since this is internal, is there any way I can actually use this?

I'm assuming its not critical for this PR, but I think exposing a basic List/Get from here is pretty reasonable.